addons/arsenal/functions/fnc_scanConfig.sqf

```diff
@@ -0,0 +1,195 @@
+/*
+ * Author: Dedmen
+ * Cache an array of all the compatible items for arsenal.
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+*/
+#include "script_component.hpp"
+
+private _cargo = [
+    [[], [], []], // Weapons 0, primary, secondary, handgun
+    [[], [], [], []], // WeaponAccessories 1, optic,side,muzzle,bipod
+    [ ], // Magazines 2
+    [ ], // Headgear 3
+    [ ], // Uniform 4
+    [ ], // Vest 5
+    [ ], // Backpacks 6
+    [ ], // Goggles 7
+    [ ], // NVGs 8
+    [ ], // Binoculars 9
+    [ ], // Map 10
+    [ ], // Compass 11
+    [ ], // Radio slot 12
+    [ ], // Watch slot  13
+    [ ], // Comms slot 14
+    [ ], // WeaponThrow 15
+    [ ], // WeaponPut 16
+    [ ] // InventoryItems 17
+];
+
+private _configCfgWeapons = configFile >> "CfgWeapons"; //Save this lookup in variable for perf improvement
+private _magazineGroups = [[],[]] call CBA_fnc_hashCreate;
+
+{
+    private _configItemInfo = _x >> "ItemInfo";
+    private _simulationType = getText (_x >> "simulation");
+    private _className = configName _x;
+
+    switch true do {
+        /* Weapon acc */
+        case (
+                isClass (_configItemInfo) &&
+                {(getNumber (_configItemInfo >> "type")) in [101, 201, 301, 302]} &&
+                {!(configName _x isKindOf ["CBA_MiscItem", (_configCfgWeapons)])}
+            ): {
+
+            switch (getNumber (_configItemInfo >> "type")) do {
```

Comment:
again `_index = [201,301,101,302] find _type`
10 commands vs... 1.

Reply:
This func only runs once per game start. So you can ignore perf improvements in here if you want to.